src/GogeDao.sol

```diff
@@ -370,718 +476,955 @@ contract GogeDAO is Ownable {
+
+            _removePollFromActivePolls(_pollNum);
+            _refundVotersPostChange(_pollNum);
+
+            _changeGovernanceToken(updateGovernanceToken.addr);
+            emit ProposalPassed(_pollNum);
+
+            return;
+        }
+
+        // remove poll from active polls and refund voters
+        _removePollFromActivePolls(_pollNum);
+        _refundVotersPostChange(_pollNum);
+
+        emit ProposalPassed(_pollNum);
+    }
+
+    /// @notice Internal method for adding voters to voterLibrary.
+    /// @param  _pollNum unique identifier for a poll.
+    /// @param  _voter address to add to voter library array.
+    function _addToVoterLibrary(uint256 _pollNum, address _voter) internal {
+        uint256 i = 0;
+        bool exists;
+        for (; i < voterLibrary[_pollNum].length; i++) {
+            if (_voter == voterLibrary[_pollNum][i]) {
+                exists = true;
+                break;
+            }
+        }
+        if (!exists) {
+            voterLibrary[_pollNum].push(_voter);
+        }
+    }
+
+    /// @notice Internal method for adding polls to advocateFor.
+    /// @param  _pollNum unique identifier for a poll.
+    /// @param  _advocate address of advocate.
+    function _addToAdvocateFor(uint256 _pollNum, address _advocate) internal {
+        uint256 i = 0;
+        bool exists;
+        for (; i < advocateFor[_advocate].length; i++) {
+            if (_pollNum == advocateFor[_advocate][i]) {
+                exists = true;
+                break;
+            }
+        }
+        if (!exists) {
+            advocateFor[_advocate].push(_pollNum);
         }
     }
 
     /// @notice A method for a voter to remove their votes from a single poll.
     /// @param  _pollNum The poll number.
-    function removeVote(uint256 _pollNum) public {
-        uint256 _numVotes = polls[_pollNum][_msgSender()];
+    function _removeVote(uint256 _pollNum) internal {
+        uint256 _numVotes = polls[_pollNum][msg.sender];
         if(_numVotes > 0) {
-            polls[_pollNum][_msgSender()] = 0;
+            polls[_pollNum][msg.sender] = 0;
             totalVotes[_pollNum] -= _numVotes;
-            if (block.timestamp <= pollEndTime[_pollNum]) {
-                historicalTally[_pollNum] -= _numVotes;
-            }
-            require(IGogeERC20(governanceTokenAddr).transfer(_msgSender(), _numVotes));
+
+            _refundVoter(msg.sender, _numVotes);
+            _removeAdvocate(msg.sender, _pollNum);
         }
     }
 
-    /// @notice A method for a voter to remove their votes from all polls.
-    function removeAllVotes() public {
-        for(uint256 i=0; i<=pollNum; i++) {
-            removeVote(i);
+    /// @notice A method for all voters to be refunded after a poll that they've voted on has been passed.
+    /// @param  _pollNum The poll number.
+    function _refundVotersPostChange(uint256 _pollNum) internal {
+        for (uint256 i = 0; i < voterLibrary[_pollNum].length; i++) {
```

Comment:
Cache the length in a local variable outside of the for loop, use `++i`, and `uint256 i;`.